### PR TITLE
also define $BLAS_SHARED_LIBS & co in build environment (analogous to $BLAS_STATIC_LIBS)

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/easybuilders/easybuild-easyblocks/issues/2521 for the need for this

### CODE REVIEW

File: easybuild/tools/toolchain/linalg.py [hunk 1]
```diff
@@ -127,7 +127,9 @@ def _set_blas_variables(self):
             self.variables.nappend('LIBBLAS', self.LIB_EXTRA, position=20)
             self.variables.nappend('LIBBLAS_MT', self.LIB_EXTRA, position=20)
 
+        self.variables.join('BLAS_SHARED_LIBS', 'LIBBLAS')
         self.variables.join('BLAS_STATIC_LIBS', 'LIBBLAS')
+        self.variables.join('BLAS_MT_SHARED_LIBS', 'LIBBLAS_MT')
         self.variables.join('BLAS_MT_STATIC_LIBS', 'LIBBLAS_MT')
```
Comment (Micket) [inline easybuild/tools/toolchain/linalg.py:133]: What is happening here? This looks strange to me that shared and static libs have something in common via LIBBLAS?

File: easybuild/tools/toolchain/variables.py [hunk 2]
```diff
@@ -111,8 +111,16 @@ def change(self, separator=None, separator_begin_end=None, prefix=None, prefix_b
                 self.END.PREFIX = prefix_begin_end
 
 
+class CommaSharedLibs(LibraryList):
+    """Comma-separated list of shared libraries"""
+    SEPARATOR = ','
+
+    PREFIX = 'lib'
+    SUFFIX = '.so'
```
Comment (Micket) [inline easybuild/tools/toolchain/variables.py:119]: get_sharedlib_ext()  here

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/tools/toolchain/variables.py (hunk 2 above)
Comment (boegel) [reply easybuild/tools/toolchain/variables.py]: fixed in b2662fb1a

File: easybuild/tools/toolchain/linalg.py (hunk 1 above)
Comment (boegel) [reply easybuild/tools/toolchain/linalg.py:133]: This is strange internal stuff. The entry for `LIBBLAS_MT` is basically just a list of library names, at this point.

The values of each variable are later "formatted" according to the mapping in `easybuild/tools/toolchain/constants.py`, so the actual value you get for `$BLAS_MT_SHARED_LIBS` and `$LIBBLAS_MT` is different.

**Review by Micket — APPROVED** [review]
lgtm

**Pull request merged**